taskvine/src/manager/vine_manager.c

```diff
@@ -3261,6 +3268,9 @@ struct vine_manager *vine_ssl_create(int port, const char *key, const char *cert
 	q->next_task_id = 1;
 	
 	q->ready_list = list_create();
+	q->running_list = list_create();
```

Comment:
It seems to me that `running_list` doesn't need to be a list, because we don't really care about the order.  Tasks get added to this when they start running, and then get removed individually when they finish.  If you convert this to a `struct table *running_table` then you can avoid the cost of linear `list_remove` operations.